```{r}
rm(list=ls())
```

```{r}
library(reshape2)
library(ggpubr)
library(readr)
library(knitr)
library(gridExtra)
library(DESeq2)
library(ggrepel)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(tidyverse)
library(enrichplot)
library(clusterProfiler)
library(msigdbr)
library(GSVA)
library(ComplexHeatmap)
library(openxlsx)
library(stringr)
library(biomaRt)
library(msigdbr)
library(ReactomePA)
library(coin)
library(glmGamPoi)
library(circlize)
```

```{r}
rcounts <- read.csv("/Users/slothking/Desktop/Krappmann3/rcounts_merged.csv")
rownames(rcounts) <- rcounts$Row.names
rcounts <- rcounts %>% dplyr::select(-c(X, Row.names))
```


```{r}
cnt <- rcounts[(rowSums(rcounts)>0),] 
```

```{r}
genemap <- readRDS("/Users/slothking/Desktop/Krappmann3/genemap.RDS")

genemap <- genemap %>%
  dplyr::group_by(hgnc_symbol) %>%
  dplyr::mutate(n_unique_ensembl_id = n()) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(unique_gene_id = case_when(
    n_unique_ensembl_id > 1 ~ paste0(hgnc_symbol, "_", ensembl_gene_id_version), 
    .default = hgnc_symbol),
    ensembl_gene_id = str_remove(ensembl_gene_id_version, "\\..*$"))
```

```{r}
metadata <- read.xlsx("/Users/slothking/Desktop/Krappmann3/metadata.xlsx")
#We have two different sequencing rounds with the same samples. The rounds are annotated as 0207 and 0301. Make the coldata:
coldata_0207 <- metadata %>% dplyr::mutate(Sample_ID = paste0(str_replace_all(Sample_ID, "ample_24L0039", ""), "_0207"),
                                           Batch = "0207")
coldata_0301 <- metadata %>% dplyr::mutate(Sample_ID = paste0(str_replace_all(Sample_ID, "ample_24L0039", ""), "_0301"),
                                           Batch = "0301")
coldata <- bind_rows(coldata_0207, coldata_0301)
```

```{r}
coldata <- subset(coldata, Sample_Group %in% c("WT_TNF_UT", "WT_TNF_1H", 
                                               "EA_TNF_UT", "EA_TNF_1H",
                                               "CAEA_TNF_UT", "CAEA_TNF_1H"))

coldata <- subset(coldata, !Sample_ID %in% c("S55_0207", "S55_0301"))

coldata <- coldata %>% 
  dplyr::arrange(desc(Sample_Group)) %>%
  dplyr::group_by(Sample_Group) %>%
  dplyr::arrange(Sample_ID) %>%
  dplyr::ungroup() %>%
  dplyr::mutate(
    Genotype =
    case_when(Sample_Group %in% c("WT_TNF_UT", "WT_TNF_1H") ~ "WT",
              Sample_Group %in% c("EA_TNF_UT", "EA_TNF_1H") ~ "EA",
              Sample_Group %in% c("CAEA_TNF_UT", "CAEA_TNF_1H") ~ "CAEA",
              TRUE ~ "other"),
    Treatment =
    case_when(Sample_Group %in% c("WT_TNF_1H", "EA_TNF_1H", "CAEA_TNF_1H") ~ "TNF",
              Sample_Group %in% c("WT_TNF_UT", "EA_TNF_UT", "CAEA_TNF_UT") ~ "UT",
              TRUE ~ "other"),
    MutEA =
    case_when(Sample_Group %in% c("EA_TNF_1H", "EA_TNF_UT", "CAEA_TNF_1H", "CAEA_TNF_UT") ~ "EAmut",
              Sample_Group %in% c("WT_TNF_UT", "WT_TNF_1H") ~ "EAwt",
              TRUE ~ "other"),
    MutCA =
    case_when(Sample_Group %in% c("CAEA_TNF_1H", "CAEA_TNF_UT") ~ "CAmut",
              Sample_Group %in% c("WT_TNF_UT", "WT_TNF_1H", "EA_TNF_UT", "EA_TNF_1H") ~ "CAwt",
              TRUE ~ "other"),)

rownames(coldata) <- coldata$Sample_ID

cnt <- dplyr::select(cnt, coldata$Sample_ID)
```

```{r}
cnt <- cnt[, rownames(coldata)]
all(rownames(coldata) %in% colnames(cnt))
all(rownames(coldata) == colnames(cnt))
```

```{r}
#Add the batch information as a confounder in the design parameter. This will account for the batch effect while calculating the differential expression analysis stats but won't normalize/transform the counts for the batch effect. We will do this later below.
dds <- DESeqDataSetFromMatrix(countData = cnt,
                                   colData = coldata,
                                   design = ~Batch+Treatment+Genotype+Genotype:Treatment)
dds_direct <- DESeqDataSetFromMatrix(countData = cnt,
                                   colData = coldata,
                                   design = ~Batch+Sample_Group)

dds$Genotype<-relevel(dds$Genotype,ref="WT")
dds$Treatment<-relevel(dds$Treatment,ref="UT")

dds <- DESeq(dds, test = "Wald")
dds_direct <- DESeq(dds_direct, test = "Wald")

# OTHER POSSIBLE TESTS:
# dds_tnf <- DESeq(dds_main, test = "LRT", reduced = ~Batch+MutEA+MutCA)
# dds_ea <- DESeq(dds_main, test = "LRT", reduced = ~Batch+Treatment+MutCA+MutCA:Treatment)
# dds_ca <- DESeq(dds_main, test = "LRT", reduced = ~Batch+Treatment+MutEA+MutEA:Treatment)
# dds_ea_tnf <- DESeq(dds_main, test = "LRT", reduced = ~Batch+Treatment+MutEA+MutCA+MutCA:Treatment)
# dds_ca_tnf <- DESeq(dds_main, test = "LRT", reduced = ~Batch+Treatment+MutEA+MutCA+MutEA:Treatment)
#dds_genotype <- DESeq(dds_main2, test = "LRT", reduced = ~Batch+Treatment+Genotype)

```
```{r}
#Apply variance stabilizing transformation on the counts
vsd <- vst(dds, blind=FALSE)
vsdMat <- assay(vsd)
```

```{r}
#Calculate and visualize PCA with the batch effect

pcaData <- plotPCA(vsd, intgroup=c("Sample_Group", "Batch"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))


pdf(file="TNFdataset/PCA_vst_WithBatchEffect_SampleGroups_TNFdataset.pdf", height=5, width=9)
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=7) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 20))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 20))+
  ggtitle("PCA With Batch Effect (TNF Dataset - Sample Groups)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=20))+
   scale_color_manual(values=c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080"))
dev.off()

pdf(file="TNFdataset/PCA_vst_WithBatchEffect_Batches_TNFdataset.pdf", height=5, width=9)
ggplot(pcaData, aes(PC1, PC2, color=Batch)) +
  geom_point(size=7) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 20))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 20))+
  ggtitle("PCA With Batch Effect (TNF Dataset - Batches)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=20))
dev.off()
```

```{r}
#Correct for the batch effect on the variance stabilized counts for downstream analyses and count visualizations

assay(vsd) <- limma::removeBatchEffect(assay(vsd),
  batch = colData(dds)[,'Batch'])

vsdmelted_temp <- melt(vsdMat)
colnames(vsdmelted_temp) <- c("ensembl_gene_id_version", "Sample_ID", "Count")

vsdmelted <- merge(vsdmelted_temp, genemap, by = "ensembl_gene_id_version")
```

```{r}
#Calculate and visualize PCA after removing the batch effect

pcaData <- plotPCA(vsd, intgroup=c("Sample_Group", "Batch"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))


pdf(file="TNFdataset/PCA_vst_BatchRemoved_SampleGroups_TNFdataset.pdf", height=5, width=9)
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=7) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 20))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 20))+
  ggtitle("PCA Without Batch Effect (TNF Dataset - Sample Groups)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=20))+
   scale_color_manual(values=c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080"))
dev.off()

pdf(file="TNFdataset/PCA_vst_BatchRemoved_Batches_TNFdataset.pdf", height=5, width=9)
ggplot(pcaData, aes(PC1, PC2, color=Batch)) +
  geom_point(size=7) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 20))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 20))+
  ggtitle("PCA Without Batch Effect (TNF Dataset - Batches)") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=20))
dev.off()
```
### TNF Main Effect

```{r}
res_TNFvsUT <- DESeq2::results(dds, contrast=c("Treatment","TNF","UT"), alpha = 0.05)
summary(res_TNFvsUT)

TNFvsUT_list<-data.frame(gene = rownames(res_TNFvsUT), res_TNFvsUT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

TNFvsUT_pos <- subset(TNFvsUT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

TNFvsUT_neg <- subset(TNFvsUT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

TNFvsUT_list <- bind_rows(TNFvsUT_pos, TNFvsUT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(TNFvsUT_list, "TNFdataset/TNFvsUT.xlsx")
```
### Mutation Main Effect

```{r}
res_EAvsWT <- DESeq2::results(dds, contrast=c("Genotype","EA","WT"), alpha = 0.05)
summary(res_EAvsWT)

EAvsWT_list<-data.frame(gene = rownames(res_EAvsWT), res_EAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

EAvsWT_pos <- subset(EAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

EAvsWT_neg <- subset(EAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

EAvsWT_list <- bind_rows(EAvsWT_pos, EAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(EAvsWT_list, "TNFdataset/EAvsWT.xlsx")
```

```{r}
res_CAEAvsWT <- DESeq2::results(dds, contrast=c("Genotype","CAEA","WT"), alpha = 0.05)
summary(res_CAEAvsWT)

CAEAvsWT_list<-data.frame(gene = rownames(res_CAEAvsWT), res_CAEAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

CAEAvsWT_pos <- subset(CAEAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

CAEAvsWT_neg <- subset(CAEAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

CAEAvsWT_list <- bind_rows(CAEAvsWT_pos, CAEAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(CAEAvsWT_list, "TNFdataset/CAEAvsWT.xlsx")
```

```{r}
res_EAvsCAEA <- DESeq2::results(dds, contrast=c("Genotype","EA","CAEA"), alpha = 0.05)
summary(res_EAvsCAEA)

EAvsCAEA_list<-data.frame(gene = rownames(res_EAvsCAEA), res_EAvsCAEA, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

EAvsCAEA_pos <- subset(EAvsCAEA_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

EAvsCAEA_neg <- subset(EAvsCAEA_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

EAvsCAEA_list <- bind_rows(EAvsCAEA_pos, EAvsCAEA_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(EAvsCAEA_list, "TNFdataset/EAvsCAEA.xlsx")
```
### EA-CAEA Signatures ###

```{r}
### EA-CAEA Signatures ###
EAvsCAEA_UP <- subset(EAvsCAEA_list, padj < 0.001 & log2FoldChange >= 0.5) %>% dplyr::arrange(desc(minuslogp))
EAvsCAEA_DN <- subset(EAvsCAEA_list, padj < 0.001 & log2FoldChange <= -0.5) %>% dplyr::arrange(desc(minuslogp))

# Create a new workbook
wb <- createWorkbook()

addWorksheet(wb, "EAvsCAEA_UP")
addWorksheet(wb, "EAvsCAEA_DN")

# Write the data to the sheet
writeData(wb, sheet = "EAvsCAEA_UP", EAvsCAEA_UP)
writeData(wb, sheet = "EAvsCAEA_DN", EAvsCAEA_DN)


  
# Optionally, you can set column widths if needed
setColWidths(wb, sheet = "EAvsCAEA_UP", cols = 1:ncol(EAvsCAEA_UP), widths = "auto")
setColWidths(wb, sheet = "EAvsCAEA_DN", cols = 1:ncol(EAvsCAEA_DN), widths = "auto")

# Save the workbook
saveWorkbook(wb, "Signatures_CAEAvsEA.xlsx", overwrite = TRUE)
```

### Interaction Effects

```{r}
res_EATNF <- DESeq2::results(dds, name="TreatmentTNF.GenotypeEA", alpha = 0.05)
summary(res_EATNF)

EATNF_list<-data.frame(gene = rownames(res_EATNF), res_EATNF, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj", "pvalue") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                signed_p = sign(log2FoldChange)*-log10(pvalue),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_p)) 

EATNF_pos <- subset(EATNF_list, signed_padj > 0) %>%
  arrange(signed_p, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

EATNF_neg <- subset(EATNF_list, signed_padj < 0) %>%
  arrange(desc(signed_p), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

EATNF_list <- bind_rows(EATNF_pos, EATNF_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(EATNF_list, "TNFdataset/EATNF.xlsx")
```
```{r}
res_CAEATNF <- DESeq2::results(dds, name="TreatmentTNF.GenotypeCAEA", alpha = 0.05)
summary(res_CAEATNF)

CAEATNF_list<-data.frame(gene = rownames(res_CAEATNF), res_CAEATNF, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj", "pvalue") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                signed_p = sign(log2FoldChange)*-log10(pvalue),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

CAEATNF_pos <- subset(CAEATNF_list, signed_padj > 0) %>%
  arrange(signed_p, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

CAEATNF_neg <- subset(CAEATNF_list, signed_padj < 0) %>%
  arrange(desc(signed_p), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

CAEATNF_list <- bind_rows(CAEATNF_pos, CAEATNF_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(CAEATNF_list, "TNFdataset/CAEATNF.xlsx")
```
### Direct Comparisons

```{r}
res_WT_TNFvsUT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","WT_TNF_1H","WT_TNF_UT"), alpha = 0.05)
summary(res_WT_TNFvsUT)

WT_TNFvsUT_list<-data.frame(gene = rownames(res_WT_TNFvsUT), res_WT_TNFvsUT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

WT_TNFvsUT_pos <- subset(WT_TNFvsUT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

WT_TNFvsUT_neg <- subset(WT_TNFvsUT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

WT_TNFvsUT_list <- bind_rows(WT_TNFvsUT_pos, WT_TNFvsUT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(WT_TNFvsUT_list, "TNFdataset/WT_TNFvsUT.xlsx")
```

```{r}
res_EA_TNFvsUT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","EA_TNF_1H","EA_TNF_UT"), alpha = 0.05)
summary(res_EA_TNFvsUT)

EA_TNFvsUT_list<-data.frame(gene = rownames(res_EA_TNFvsUT), res_EA_TNFvsUT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

EA_TNFvsUT_pos <- subset(EA_TNFvsUT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

EA_TNFvsUT_neg <- subset(EA_TNFvsUT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

EA_TNFvsUT_list <- bind_rows(EA_TNFvsUT_pos, EA_TNFvsUT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(EA_TNFvsUT_list, "TNFdataset/EA_TNFvsUT.xlsx")
```
```{r}
res_CAEA_TNFvsUT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","CAEA_TNF_1H","CAEA_TNF_UT"), alpha = 0.05)
summary(res_CAEA_TNFvsUT)

CAEA_TNFvsUT_list<-data.frame(gene = rownames(res_CAEA_TNFvsUT), res_CAEA_TNFvsUT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

CAEA_TNFvsUT_pos <- subset(CAEA_TNFvsUT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

CAEA_TNFvsUT_neg <- subset(CAEA_TNFvsUT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

CAEA_TNFvsUT_list <- bind_rows(CAEA_TNFvsUT_pos, CAEA_TNFvsUT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(CAEA_TNFvsUT_list, "TNFdataset/CAEA_TNFvsUT.xlsx")
```

```{r}
res_TNF_EAvsWT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","EA_TNF_1H","WT_TNF_1H"), alpha = 0.05)
summary(res_TNF_EAvsWT)

TNF_EAvsWT_list<-data.frame(gene = rownames(res_TNF_EAvsWT), res_TNF_EAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

TNF_EAvsWT_pos <- subset(TNF_EAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

TNF_EAvsWT_neg <- subset(TNF_EAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

TNF_EAvsWT_list <- bind_rows(TNF_EAvsWT_pos, TNF_EAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(TNF_EAvsWT_list, "TNFdataset/TNF_EAvsWT.xlsx")
```

```{r}
res_TNF_CAEAvsWT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","CAEA_TNF_1H","WT_TNF_1H"), alpha = 0.05)
summary(res_TNF_CAEAvsWT)

TNF_CAEAvsWT_list<-data.frame(gene = rownames(res_TNF_CAEAvsWT), res_TNF_CAEAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

TNF_CAEAvsWT_pos <- subset(TNF_CAEAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

TNF_CAEAvsWT_neg <- subset(TNF_CAEAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

TNF_CAEAvsWT_list <- bind_rows(TNF_CAEAvsWT_pos, TNF_CAEAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(TNF_CAEAvsWT_list, "TNFdataset/TNF_CAEAvsWT.xlsx")
```
```{r}
res_TNF_CAEAvsEA <- DESeq2::results(dds_direct, contrast=c("Sample_Group","CAEA_TNF_1H","EA_TNF_1H"), alpha = 0.05)
summary(res_TNF_CAEAvsEA)

TNF_CAEAvsEA_list<-data.frame(gene = rownames(res_TNF_CAEAvsEA), res_TNF_CAEAvsEA, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

TNF_CAEAvsEA_pos <- subset(TNF_CAEAvsEA_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

TNF_CAEAvsEA_neg <- subset(TNF_CAEAvsEA_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

TNF_CAEAvsEA_list <- bind_rows(TNF_CAEAvsEA_pos, TNF_CAEAvsEA_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(TNF_CAEAvsEA_list, "TNFdataset/TNF_CAEAvsEA.xlsx")
```

```{r}
res_UT_EAvsWT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","EA_TNF_UT","WT_TNF_UT"), alpha = 0.05)
summary(res_UT_EAvsWT)

UT_EAvsWT_list<-data.frame(gene = rownames(res_UT_EAvsWT), res_UT_EAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

UT_EAvsWT_pos <- subset(UT_EAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

UT_EAvsWT_neg <- subset(UT_EAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

UT_EAvsWT_list <- bind_rows(UT_EAvsWT_pos, UT_EAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(UT_EAvsWT_list, "TNFdataset/UT_EAvsWT.xlsx")
```
```{r}
res_UT_CAEAvsWT <- DESeq2::results(dds_direct, contrast=c("Sample_Group","CAEA_TNF_UT","WT_TNF_UT"), alpha = 0.05)
summary(res_UT_CAEAvsWT)

UT_CAEAvsWT_list<-data.frame(gene = rownames(res_UT_CAEAvsWT), res_UT_CAEAvsWT, stringsAsFactors = F) %>% 
  na.omit() %>% 
  dplyr::select("log2FoldChange", "padj") %>% 
  tibble::rownames_to_column(var = "ensembl_gene_id_version") %>% 
  dplyr::mutate(padj = case_when(padj == 0 ~ .Machine$double.xmin,
                                TRUE ~ padj), 
                signed_padj = sign(log2FoldChange)*-log10(padj),
                minuslogp = -log10(padj)) %>%
  merge(genemap, by="ensembl_gene_id_version") %>%
  dplyr::arrange(desc(signed_padj)) 

UT_CAEAvsWT_pos <- subset(UT_CAEAvsWT_list, signed_padj > 0) %>%
  arrange(signed_padj, abs(log2FoldChange)) %>%
  mutate(rank = row_number()) %>%
  arrange(desc(rank))

UT_CAEAvsWT_neg <- subset(UT_CAEAvsWT_list, signed_padj < 0) %>%
  arrange(desc(signed_padj), abs(log2FoldChange)) %>%
  mutate(rank = -row_number()) %>%
  arrange(desc(rank))

UT_CAEAvsWT_list <- bind_rows(UT_CAEAvsWT_pos, UT_CAEAvsWT_neg) %>% dplyr::arrange(desc(rank))

write.xlsx(UT_CAEAvsWT_list, "TNFdataset/UT_CAEAvsWT.xlsx")
```
### Signed Log-padj Plots

```{r}
upreg_signlog_cutoff = -log10(0.05)
downreg_signlog_cutoff = log10(0.05)
significance_p_cutoff = 0.05 

TNF_CAEA_EA_merged <- merge(TNF_CAEAvsWT_list, TNF_EAvsWT_list, by="hgnc_symbol", suffixes = c(".CAEA",".EA"), all=TRUE) %>% 
  dplyr::mutate(
    signed_padj.CAEA = case_when(is.na(signed_padj.CAEA) ~ 0,
                                 TRUE ~ signed_padj.CAEA),
    signed_padj.EA = case_when(is.na(signed_padj.EA) ~ 0,
                               TRUE ~ signed_padj.EA),
    padj.CAEA = case_when(is.na(padj.CAEA) ~ 1,
                                 TRUE ~ padj.CAEA),
    padj.EA = case_when(is.na(padj.EA) ~ 1,
                                 TRUE ~ padj.EA),
    significance = case_when(signed_padj.EA > upreg_signlog_cutoff & padj.CAEA > significance_p_cutoff ~ "EAup",
                             signed_padj.EA < downreg_signlog_cutoff & padj.CAEA > significance_p_cutoff ~ "EAdown",
                             padj.EA > significance_p_cutoff & signed_padj.CAEA > upreg_signlog_cutoff ~ "CAEAup",
                             padj.EA > significance_p_cutoff & signed_padj.CAEA < downreg_signlog_cutoff ~ "CAEAdown",
                             signed_padj.EA > upreg_signlog_cutoff & signed_padj.CAEA > upreg_signlog_cutoff ~ "BOTHup",
                             signed_padj.EA < downreg_signlog_cutoff & signed_padj.CAEA < downreg_signlog_cutoff ~ "BOTHdown",
                             TRUE ~ NA_character_))

# Prepare data for labeling
EAup_EAdown_top <- TNF_CAEA_EA_merged %>%
  filter(significance %in% c("EAup", "EAdown")) %>%
  arrange(desc(abs(signed_padj.EA))) %>%
  slice(1:10)

CAEAup_CAEAdown_top <- TNF_CAEA_EA_merged %>%
  filter(significance %in% c("CAEAup", "CAEAdown")) %>%
  arrange(desc(abs(signed_padj.CAEA))) %>%
  slice(1:10)

BOTHup_BOTHdown_top <- TNF_CAEA_EA_merged %>%
  filter(significance %in% c("BOTHup", "BOTHdown")) %>%
  mutate(mean_abs_padj = (abs(signed_padj.CAEA) + abs(signed_padj.EA))/2) %>%
  arrange(desc(mean_abs_padj)) %>%
  slice(1:10)


pdf(file="TNFdataset/signlog/signlog_TNF_CAEA_EA.pdf", width=8, height=8)
ggplot(TNF_CAEA_EA_merged, aes(x=signed_padj.CAEA, y=signed_padj.EA))+
  geom_point(color="grey60", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  geom_point(data=subset(TNF_CAEA_EA_merged, significance %in% c("EAup", "EAdown")),
                  aes(label=hgnc_symbol),
                  color="darkred",
                  alpha=0.2) +
  geom_point(data=subset(TNF_CAEA_EA_merged, significance %in% c("CAEAup", "CAEAdown")),
                  aes(label=hgnc_symbol),
                  color="darkblue", 
                  alpha=0.2) +
  geom_point(data=subset(TNF_CAEA_EA_merged, significance %in% c("BOTHup", "BOTHdown")),
                  aes(label=hgnc_symbol),
                  color="purple",
                  alpha=0.2) +
  geom_text_repel(data = EAup_EAdown_top, aes(label = hgnc_symbol), color = "darkred") +
  geom_text_repel(data = CAEAup_CAEAdown_top, aes(label = hgnc_symbol), color = "darkblue") +
  geom_text_repel(data = BOTHup_BOTHdown_top, aes(label = hgnc_symbol), color = "purple", max.overlaps = Inf)
dev.off()

write.xlsx(TNF_CAEA_EA_merged, "TNFdataset/signlog/TNF_CrossComparison.xlsx")
```

```{r}
UT_CAEA_EA_merged <- merge(UT_CAEAvsWT_list, UT_EAvsWT_list, by="hgnc_symbol", suffixes = c(".CAEA",".EA"), all=TRUE) %>% 
  dplyr::mutate(
    signed_padj.CAEA = case_when(is.na(signed_padj.CAEA) ~ 0,
                                 TRUE ~ signed_padj.CAEA),
    signed_padj.EA = case_when(is.na(signed_padj.EA) ~ 0,
                               TRUE ~ signed_padj.EA),
    padj.CAEA = case_when(is.na(padj.CAEA) ~ 1,
                                 TRUE ~ padj.CAEA),
    padj.EA = case_when(is.na(padj.EA) ~ 1,
                                 TRUE ~ padj.EA),
    significance = case_when(signed_padj.EA > upreg_signlog_cutoff & padj.CAEA > significance_p_cutoff ~ "EAup",
                             signed_padj.EA < downreg_signlog_cutoff & padj.CAEA > significance_p_cutoff ~ "EAdown",
                             padj.EA > significance_p_cutoff & signed_padj.CAEA > upreg_signlog_cutoff ~ "CAEAup",
                             padj.EA > significance_p_cutoff & signed_padj.CAEA < downreg_signlog_cutoff ~ "CAEAdown",
                             signed_padj.EA > upreg_signlog_cutoff & signed_padj.CAEA > upreg_signlog_cutoff ~ "BOTHup",
                             signed_padj.EA < downreg_signlog_cutoff & signed_padj.CAEA < downreg_signlog_cutoff ~ "BOTHdown",
                             TRUE ~ NA_character_))

# Prepare data for labeling
EAup_EAdown_top <- UT_CAEA_EA_merged %>%
  filter(significance %in% c("EAup", "EAdown")) %>%
  arrange(desc(abs(signed_padj.EA))) %>%
  slice(1:10)

CAEAup_CAEAdown_top <- UT_CAEA_EA_merged %>%
  filter(significance %in% c("CAEAup", "CAEAdown")) %>%
  arrange(desc(abs(signed_padj.CAEA))) %>%
  slice(1:10)

BOTHup_BOTHdown_top <- UT_CAEA_EA_merged %>%
  filter(significance %in% c("BOTHup", "BOTHdown")) %>%
  mutate(mean_abs_padj = (abs(signed_padj.CAEA) + abs(signed_padj.EA))/2) %>%
  arrange(desc(mean_abs_padj)) %>%
  slice(1:10)


pdf(file="TNFdataset/signlog/signlog_UT_CAEA_EA.pdf", width=8, height=8)
ggplot(UT_CAEA_EA_merged, aes(x=signed_padj.CAEA, y=signed_padj.EA))+
  geom_point(color="grey60", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  geom_point(data=subset(UT_CAEA_EA_merged, significance %in% c("EAup", "EAdown")),
                  aes(label=hgnc_symbol),
                  color="darkred",
                  alpha=0.2) +
  geom_point(data=subset(UT_CAEA_EA_merged, significance %in% c("CAEAup", "CAEAdown")),
                  aes(label=hgnc_symbol),
                  color="darkblue", 
                  alpha=0.2) +
  geom_point(data=subset(UT_CAEA_EA_merged, significance %in% c("BOTHup", "BOTHdown")),
                  aes(label=hgnc_symbol),
                  color="purple",
                  alpha=0.2) +
  geom_text_repel(data = EAup_EAdown_top, aes(label = hgnc_symbol), color = "darkred") +
  geom_text_repel(data = CAEAup_CAEAdown_top, aes(label = hgnc_symbol), color = "darkblue") +
  geom_text_repel(data = BOTHup_BOTHdown_top, aes(label = hgnc_symbol), color = "purple", max.overlaps = Inf)
dev.off()

write.xlsx(UT_CAEA_EA_merged, "TNFdataset/signlog/UT_CrossComparison.xlsx")
```

```{r}
CAEA_EA_merged <- merge(CAEAvsWT_list, EAvsWT_list, by="hgnc_symbol", suffixes = c(".CAEA",".EA"), all=TRUE) %>% 
  dplyr::mutate(
    signed_padj.CAEA = case_when(signed_padj.CAEA == NA ~ 0,
                                 .default = signed_padj.CAEA),
    signed_padj.EA = case_when(signed_padj.EA == NA ~ 0,
                               .default = signed_padj.EA)
  )

CAEAonly_genes_df <- subset(CAEA_EA_merged, padj.EA > 0.001) %>% subset(padj.CAEA < 0.001) %>% dplyr::arrange(desc(minuslogp.CAEA))
EAonly_genes_df <- subset(CAEA_EA_merged, padj.CAEA > 0.001) %>% subset(padj.EA < 0.001) %>% dplyr::arrange(desc(minuslogp.EA))
  
   
pdf(file="TNFdataset/signlog/signlog_Genotype_CAEA_EA.pdf", width=8, height=8)
ggplot(CAEA_EA_merged, aes(x=signed_padj.CAEA, y=signed_padj.EA))+
  geom_point(color="grey80", alpha=0.6)+
  geom_point(data=EAonly_genes_df, color="#ADD8E6")+
  geom_point(data=CAEAonly_genes_df, color="#DDA0DD")+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  geom_text_repel(data=head(EAonly_genes_df,n=10), 
                  aes(label=hgnc_symbol),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)+
  geom_text_repel(data=head(CAEAonly_genes_df,n=10), 
                  aes(label=hgnc_symbol),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)+
    xlab("sign(log2FoldChange)x(-log10padj) [CAEA]")+
    ylab("sign(log2FoldChange)x(-log10padj) [EA]")
dev.off()
```

```{r}
# selected_genes <- unique(intersect(
#   intersect(subset(wt_TNFvsUT_list, abs(log2FoldChange) >= 0.5 & padj <= 0.05)$ensembl_gene_id_version,
#             subset(ea_TNFvsUT_list, abs(log2FoldChange) >= 0.5 & padj <= 0.05)$ensembl_gene_id_version),
#   subset(caea_TNFvsUT_list, abs(log2FoldChange) >= 0.5 & padj <= 0.05)$ensembl_gene_id_version))

tnfUP_signature <- subset(TNFvsUT_list, padj <= 0.001 & log2FoldChange >= 0.5)$ensembl_gene_id_version
saveRDS(tnfUP_signature, "tnfUP_signature.RDS")
#EAvsWT_list_subset <- subset(EAvsWT_list, padj <= 0.001)$ensembl_gene_id_version

#overlap <- intersect(tnfUP_signature, EAvsWT_list_subset)
#selected_genes <- tnfUP_signature
```

### VOLCANO PLOTS ###

```{r}
# TNF main effect Volcano Plot

pdf(file="TNFdataset/VolcanoPlots/Volcano_TNFvsUT.pdf", width=6, height=6)
ggplot(TNFvsUT_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(TNFvsUT_list, padj < 0.001), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,325,by=25), limits = c(0,325))+
  scale_x_continuous(breaks = seq(-3,3,by=1), limits = c(-3,4))+
  geom_text_repel(data=subset(TNFvsUT_list, ensembl_gene_id_version %in% tnfUP_signature), aes(label=unique_gene_id),
                  size=1.5,
                  color="black",
                  max.overlaps = Inf,
                  segment.color = "grey80")
dev.off()
```

```{r}
# EA vs WT Volcano Plot

pdf(file="TNFdataset/VolcanoPlots/Volcano_EAvsWT.pdf", width=6, height=6)
ggplot(EAvsWT_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(EAvsWT_list, padj < 0.001 & abs(log2FoldChange) > 0.5), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,40,by=5), limits = c(0,40))+
  scale_x_continuous(breaks = seq(-3.8,2.2,by=0.5), limits = c(-3.8,2.2))+
  geom_text_repel(data=subset(EAvsWT_list, minuslogp > 12), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf) +
  geom_text_repel(data=subset(EAvsWT_list, log2FoldChange < -2), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf) +
    geom_text_repel(data=subset(EAvsWT_list, log2FoldChange > 1.5), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)
dev.off()
```

```{r}
# CAEA vs WT Volcano Plot

pdf(file="TNFdataset/VolcanoPlots/Volcano_CAEAvsWT.pdf", width=6, height=6)
ggplot(CAEAvsWT_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(CAEAvsWT_list, padj < 0.001, abs(log2FoldChange) > log2(1.5)), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,135,by=15), limits = c(0,130))+
  scale_x_continuous(breaks = seq(-5.8,2.2,by=0.5), limits = c(-5.8,2.2))+
  geom_text_repel(data=subset(CAEAvsWT_list, minuslogp > 50), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)
dev.off()
```

```{r}
# CAEA vs EA Volcano Plot

# pdf(file="TNFdataset/VolcanoPlots/Volcano_EAvsCAEA.pdf", width=6, height=6)
# ggplot(EAvsCAEA_list, aes(x=log2FoldChange, y=(minuslogp)))+
#   geom_point(size=2,color="grey", alpha=0.3)+
#   geom_point(data=subset(CAEAvsEA_list, padj < 0.001 & abs(log2FoldChange) > 0.5), size=2,color="darkred", alpha=0.4)+
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
#         panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
#   xlab("log2FoldChange")+
#   theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
#   ylab("-log10(p-adj)")+
#   theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
#   scale_y_continuous(breaks = seq(0,135,by=15), limits = c(0,130))+
#   scale_x_continuous(breaks = seq(-3,2,by=0.5), limits = c(-3,2))+
#   geom_text_repel(data=subset(CAEAvsEA_list, padj < 0.00001 & log2FoldChange > 0.5), aes(label=unique_gene_id),
#                   size=3,
#                   color="black",
#                   direction = "y",
#                   nudge_x = 1.2,
#                   force=10,
#                   min.segment.length = 0,
#                   max.overlaps = Inf)+
#   geom_text_repel(data=subset(CAEAvsEA_list, padj < 0.00001 & log2FoldChange < -0.5), aes(label=unique_gene_id),
#                   size=3,
#                   color="black",
#                   direction = "y",
#                   nudge_x = -1.2,
#                   force=10,
#                   min.segment.length = 0,
#                   max.overlaps = Inf)
# dev.off()
```

```{r}
# EA-TNF interaction effect Volcano Plot

pdf(file="TNFdataset/VolcanoPlots/Volcano_EATNFinteraction.pdf", width=6, height=6)
ggplot(EATNF_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(EATNF_list, padj < 0.001), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,10,by=1), limits = c(0,10))+
  scale_x_continuous(breaks = seq(-5,5,by=1), limits = c(-5,5))
dev.off()
```

```{r}
# CA-TNF interaction effect Volcano Plot

pdf(file="TNFdataset/VolcanoPlots/Volcano_CAEATNFinteraction.pdf", width=6, height=6)
ggplot(CAEATNF_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(CAEATNF_list, padj < 0.001), size=2,color="darkred", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,10,by=1), limits = c(0,10))+
  scale_x_continuous(breaks = seq(-5,5,by=1), limits = c(-5,5))
dev.off()
```

## Direct Comparison Volcano Plots

```{r}
# EA mutated TNF treatment vs WT TNF treatment

pdf(file="TNFdataset/VolcanoPlots/Volcano_TNF_EAvsWT.pdf", width=6, height=6)
ggplot(TNF_EAvsWT_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(TNF_EAvsWT_list, padj<0.001), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,50,by=5), limits = c(0,50))+
  scale_x_continuous(breaks = seq(-1.4,1.4,by=0.4), limits = c(-1.4,1.4))+
  geom_text_repel(data=subset(TNF_EAvsWT_list, minuslogp > 5 & abs(log2FoldChange) > 0.2), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)
dev.off()
```

```{r}
# CAEA mutated TNF treatment vs WT TNF treatment

pdf(file="TNFdataset/VolcanoPlots/Volcano_TNF_CAEAvsWT.pdf", width=6, height=6)
ggplot(TNF_CAEAvsWT_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2,color="grey", alpha=0.3)+
  geom_point(data=subset(TNF_CAEAvsWT_list, padj < 0.001), size=2,color="darkred", alpha=0.4)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("log2FoldChange")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  scale_y_continuous(breaks = seq(0,160,by=10), limits = c(0,160))+
  scale_x_continuous(breaks = seq(-3.2,3.2,by=0.4), limits = c(-3.2,3.2))+
  geom_text_repel(data=subset(TNF_CAEAvsWT_list, minuslogp > 40), aes(label=unique_gene_id),
                  size=4,
                  color="black",
                  nudge_y = 1,
                  force=10,
                  min.segment.length = 0,
                  max.overlaps = Inf)
dev.off()
```

```{r}
tnf_signature_df <- subset(TNFvsUT_list, ensembl_gene_id_version %in% tnfUP_signature) %>%
  dplyr::select(-c(rank, n_unique_ensembl_id))

write.xlsx(tnf_signature_df, "tnf_signature.xlsx")
```

### HEATMAPS

```{r}
data_heatmap <- subset(vsdMat, rownames(vsdMat) %in% tnfUP_signature)

scaled_data = t(scale(t(data_heatmap)))

ordered_data <- as.data.frame(scaled_data[,coldata$Sample_ID])

ordered_data$ensembl_gene_id_version <- rownames(ordered_data)

ordered_data_temp <- merge(ordered_data, genemap, by="ensembl_gene_id_version")

rownames(ordered_data_temp) <-ordered_data_temp$unique_gene_id

ordered_data <- ordered_data_temp[,coldata$Sample_ID]

group_labels <- factor(coldata$Sample_Group)
group_labels_unique <- unique(group_labels)
group_colors <- c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080")
names(group_colors) <- group_labels_unique



column_annotation <- HeatmapAnnotation(
  group = group_labels,
  col = list(group = group_colors)
)

# Calculate the minimum and maximum of the transformed data
#min_val <- min(ordered_data)
#max_val <- max(ordered_data)

# Introduce an intermediate value closer to min_val where you want yellow to start appearing more prominently
# Adjust this value to control how quickly the transition to yellow occurs
#intermediate_val <- min_val + (max_val - min_val) * 0.25  # Adjust this factor based on your data

#col_fun <- colorRamp2(c(min_val, 0, max_val), c("navyblue", "grey100", "darkred"))

heatmap_colors <- colorRamp2(c(min(ordered_data), 0, max(ordered_data)),
                             c("#6495ED", "#FFFFFF", "#FF4500"))

ht = Heatmap(ordered_data,
        name = "Expression",
        show_row_names = TRUE,
        show_column_names = TRUE,
        cluster_columns = FALSE,
        cluster_rows = TRUE,
        #col = col_fun,
        row_names_gp = gpar(fontsize = 5),
        top_annotation = column_annotation,
        col = heatmap_colors,  # Apply custom colors
        column_names_gp = gpar(fontsize = 4),
        use_raster = FALSE)

ht = draw(ht)

pdf("TNFdataset/Heatmap_TNFsignature.pdf", width = 6, height = 4)
ht
dev.off()
```

```{r}
cgp_genesets <- msigdbr(species = "Homo sapiens",
                             category = "C2",
                             subcategory = "CGP") %>%
  dplyr::select(gs_name, ensembl_gene) %>% distinct()

cgp_genesets_subset <- cgp_genesets %>%
  subset(gs_name %in% c("TIAN_TNF_SIGNALING_VIA_NFKB"))

tnf_signature_gsva_df <- tnf_signature_df %>% dplyr::select(c(ensembl_gene_id)) %>%
  dplyr::rename("ensembl_gene" = "ensembl_gene_id") %>%
  dplyr::mutate(gs_name = "TNF_UP_SIGNATURE")

gsva_df <- bind_rows(cgp_genesets_subset, tnf_signature_gsva_df)
```

```{r}
genesets_list <- unlist(split(gsva_df,
                              f = gsva_df$gs_name),
                        recursive = F)

vsdMat_ensembl <- vsdMat
rownames(vsdMat_ensembl) <- str_remove(rownames(vsdMat), "\\..*$")

gsva_object <- gsvaParam(exprData = as.matrix(vsdMat_ensembl), geneSets = genesets_list)
gsva_res <- gsva(gsva_object, verbose=FALSE)

gsva_res_melted <- reshape2::melt(gsva_res, 
                                  varnames=c('GeneSet', 'Sample_ID'),
                                  value.name = "GSVAscore") %>%
  dplyr::mutate(GeneSet = str_remove_all(GeneSet, ".gene_symbol"))

gsva_res_merged <- merge(gsva_res_melted, coldata, by="Sample_ID")

gsva_res_merged$Sample_Group <- factor(gsva_res_merged$Sample_Group, levels =
                                                                   c("WT_TNF_UT", "WT_TNF_1H",
                                                                     "EA_TNF_UT","EA_TNF_1H",
                                                                     "CAEA_TNF_UT", "CAEA_TNF_1H"))

gsva_res_merged <- gsva_res_merged %>% arrange(Sample_Group) %>% dplyr::mutate(GeneSet = str_replace_all(GeneSet, ".ensembl_gene", ""))

# Define group colors
group_colors <- c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080")

# Calculate means for each SampleGroup
median <- gsva_res_merged %>%
  group_by(Sample_Group, GeneSet) %>%
  summarize(median_GSVAscore = median(GSVAscore))

pdf("TNFdataset/GSVA.pdf", width = 3, height = 8)
ggplot(gsva_res_merged, aes(y = GSVAscore, x = Sample_Group, fill = Sample_Group)) +
  geom_violin(alpha = 0.5, show.legend = FALSE) +  # Transparent violins and remove legend for fill
  geom_jitter(width = 0.15, size = 1.5, alpha = 0.6, color = "black") +  # Black jitter points with adjusted size and transparency
  scale_fill_manual(values = group_colors) +
  theme_minimal(base_size = 12) +  # Increase base font size for better readability
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12),  # Adjust x-axis text size
    axis.title.x = element_text(size = 14),  # Add and adjust x-axis title
    axis.title.y = element_text(size = 14),  # Add and adjust y-axis title
    strip.text = element_text(size = 8),  # Adjust facet label size
    legend.position = "none",  # Remove legend
    panel.spacing = unit(1, "lines")  # Increase spacing between facets
  ) +
  labs(
    x = "Sample Group",  # Add x-axis label
    y = "GSVA Score",  # Add y-axis label
  ) +
  facet_wrap(~GeneSet, ncol = 1, nrow = 2)  # Adjust number of rows and columns in facet
dev.off()
```
```{r}
# Wilcoxon test for GSVA (TIAN_TNF_SIGNALING_VIA_NFKB)
wt_TNF <- subset(gsva_res_merged, Sample_Group == "WT_TNF_1H" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore
wt_UT <- subset(gsva_res_merged, Sample_Group == "WT_TNF_UT" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore
ea_TNF <- subset(gsva_res_merged, Sample_Group == "EA_TNF_1H" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore
ea_UT <- subset(gsva_res_merged, Sample_Group == "EA_TNF_UT" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore
caea_TNF <- subset(gsva_res_merged, Sample_Group == "CAEA_TNF_1H" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore
caea_UT <- subset(gsva_res_merged, Sample_Group == "CAEA_TNF_UT" &
                            GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")$GSVAscore

res_wt_TNF <- wilcox.test(wt_TNF, wt_UT, correct = TRUE)
res_ea_TNF <- wilcox.test(ea_TNF, ea_UT, correct = TRUE)
res_caea_TNF <- wilcox.test(caea_TNF, caea_UT, correct = TRUE)
res_TNF_ea_vs_wt <- wilcox.test(ea_TNF, wt_TNF, correct = TRUE)
res_TNF_caea_vs_wt <- wilcox.test(caea_TNF, wt_TNF, correct = TRUE)
res_TNF_caea_vs_ea <- wilcox.test(caea_TNF, ea_TNF, correct = TRUE)

res_wt_TNF
res_ea_TNF
res_caea_TNF
res_TNF_ea_vs_wt
res_TNF_caea_vs_wt
res_TNF_caea_vs_ea
```

```{r}
# Wilcoxon test for GSVA (TNF_UP_SIGNATURE)
wt_TNF <- subset(gsva_res_merged, Sample_Group == "WT_TNF_1H" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore
wt_UT <- subset(gsva_res_merged, Sample_Group == "WT_TNF_UT" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore
ea_TNF <- subset(gsva_res_merged, Sample_Group == "EA_TNF_1H" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore
ea_UT <- subset(gsva_res_merged, Sample_Group == "EA_TNF_UT" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore
caea_TNF <- subset(gsva_res_merged, Sample_Group == "CAEA_TNF_1H" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore
caea_UT <- subset(gsva_res_merged, Sample_Group == "CAEA_TNF_UT" &
                            GeneSet == "TNF_UP_SIGNATURE")$GSVAscore

res_wt_TNF <- wilcox.test(wt_TNF, wt_UT, correct = TRUE)
res_ea_TNF <- wilcox.test(ea_TNF, ea_UT, correct = TRUE)
res_caea_TNF <- wilcox.test(caea_TNF, caea_UT, correct = TRUE)
res_TNF_ea_vs_wt <- wilcox.test(ea_TNF, wt_TNF, correct = TRUE)
res_TNF_caea_vs_wt <- wilcox.test(caea_TNF, wt_TNF, correct = TRUE)
res_TNF_caea_vs_ea <- wilcox.test(caea_TNF, ea_TNF, correct = TRUE)

res_wt_TNF
res_ea_TNF
res_caea_TNF
res_TNF_ea_vs_wt
res_TNF_caea_vs_wt
res_TNF_caea_vs_ea
```

```{r}
# # Independence test for GSVA (TIAN TNF SIGNALLING VIA NFKB)
# wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("WT_TNF_1H",  "WT_TNF_UT") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# ea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("EA_TNF_1H",  "EA_TNF_UT") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# caea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "CAEA_TNF_UT") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# ea_vs_wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("EA_TNF_1H",  "WT_TNF_1H") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# caea_vs_wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "WT_TNF_1H") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# caea_vs_ea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "EA_TNF_1H") &
#                                       GeneSet == "TIAN_TNF_SIGNALING_VIA_NFKB")
# 
# res_wt <- independence_test(wt$GSVAscore ~ factor(wt$Sample_Group, levels = c("WT_TNF_1H", "WT_TNF_UT")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_ea <- independence_test(ea$GSVAscore ~ factor(ea$Sample_Group, levels = c("EA_TNF_1H", "EA_TNF_UT")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea <- independence_test(caea$GSVAscore ~ factor(caea$Sample_Group, levels = c("CAEA_TNF_1H", "CAEA_TNF_UT")),
#                                                       distribution = "asymptotic",
#                                                       alternative = "two.sided")
# 
# res_ea_vs_wt <- independence_test(ea_vs_wt$GSVAscore ~ factor(ea_vs_wt$Sample_Group, levels = c("EA_TNF_1H", "WT_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea_vs_wt <- independence_test(caea_vs_wt$GSVAscore ~ factor(caea_vs_wt$Sample_Group, levels = c("CAEA_TNF_1H", "WT_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea_vs_ea <- independence_test(caea_vs_ea$GSVAscore ~ factor(caea_vs_ea$Sample_Group, levels = c("CAEA_TNF_1H", "EA_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_wt
# res_ea
# res_caea
# res_ea_vs_wt
# res_caea_vs_wt
# res_caea_vs_ea
```
```{r}
# # Independence test for GSVA (TNF_UP_SIGNATURE)
# wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("WT_TNF_1H",  "WT_TNF_UT") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# ea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("EA_TNF_1H",  "EA_TNF_UT") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# caea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "CAEA_TNF_UT") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# ea_vs_wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("EA_TNF_1H",  "WT_TNF_1H") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# caea_vs_wt <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "WT_TNF_1H") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# caea_vs_ea <- subset(gsva_res_merged, Sample_Group %in%
#                                       c("CAEA_TNF_1H",  "EA_TNF_1H") &
#                                       GeneSet == "TNF_UP_SIGNATURE")
# 
# res_wt <- independence_test(wt$GSVAscore ~ factor(wt$Sample_Group, levels = c("WT_TNF_1H", "WT_TNF_UT")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_ea <- independence_test(ea$GSVAscore ~ factor(ea$Sample_Group, levels = c("EA_TNF_1H", "EA_TNF_UT")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea <- independence_test(caea$GSVAscore ~ factor(caea$Sample_Group, levels = c("CAEA_TNF_1H", "CAEA_TNF_UT")),
#                                                       distribution = "asymptotic",
#                                                       alternative = "two.sided")
# 
# res_ea_vs_wt <- independence_test(ea_vs_wt$GSVAscore ~ factor(ea_vs_wt$Sample_Group, levels = c("EA_TNF_1H", "WT_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea_vs_wt <- independence_test(caea_vs_wt$GSVAscore ~ factor(caea_vs_wt$Sample_Group, levels = c("CAEA_TNF_1H", "WT_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_caea_vs_ea <- independence_test(caea_vs_ea$GSVAscore ~ factor(caea_vs_ea$Sample_Group, levels = c("CAEA_TNF_1H", "EA_TNF_1H")),
#                                                   distribution = "asymptotic",
#                                                   alternative = "two.sided")
# 
# res_wt
# res_ea
# res_caea
# res_ea_vs_wt
# res_caea_vs_wt
# res_caea_vs_ea
```

```{r}
discovery_genes <- c("ENSG00000105369.10")

vsd_counts_melted_selectedgenes <- subset(vsdmelted, ensembl_gene_id_version %in% discovery_genes)

vsd_counts_melted_selectedgenes_annotated <- merge(vsd_counts_melted_selectedgenes, coldata, by="Sample_ID")

vsd_counts_melted_selectedgenes_annotated$Sample_Group <- factor(vsd_counts_melted_selectedgenes_annotated$Sample_Group, levels =
                                                                   c("WT_TNF_UT", "WT_TNF_1H",
                                                                     "EA_TNF_UT","EA_TNF_1H",
                                                                     "CAEA_TNF_UT", "CAEA_TNF_1H"))

vsd_counts_melted_selectedgenes_annotated <- vsd_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)

#Not actually subsetting here because we are choosing the entire dataset but this is how to subset in case if it is needed.
vsd_counts_melted_selectedgenes_annotated_subset <- subset(vsd_counts_melted_selectedgenes_annotated, Sample_Group %in%
                                                             c("WT_TNF_UT", "WT_TNF_1H",
                                                               "EA_TNF_UT","EA_TNF_1H",
                                                               "CAEA_TNF_UT", "CAEA_TNF_1H"))


# Start a new PDF file with 1 plot per page
pdf("TNFdataset/gene_boxplots_discovery.pdf", width = 1.8, height=3)

# Loop over each gene and create a plot on a new page
for (g in unique(vsd_counts_melted_selectedgenes_annotated_subset$ensembl_gene_id_version)) {
  # Subset the data for the current gene
  gene_data <- subset(vsd_counts_melted_selectedgenes_annotated_subset, ensembl_gene_id_version == g)
  # Create the plot for the current gene
  p <- ggplot(gene_data, aes(x = Sample_Group, y = Count, color = Sample_Group)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(width = 0.1) +
    scale_color_manual(values=c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080")) +
    ylab("Normalized Count") +
    ggtitle(paste(gene_data$unique_gene_id)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          plot.background = element_blank(),
          axis.line = element_line(),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
          plot.title = element_text(size=6),
          axis.title.x = element_blank(),
          axis.title.y = element_text(size=10))+
    # theme(axis.title.x = element_blank(),
    #       axis.ticks.x = element_blank(),
    #       strip.background = element_blank(),
    #       strip.placement = "outside",
    #       strip.text = element_text(size = 10, face = "bold"),
    #       axis.text.x = element_text(angle = 90))+
    guides(color="none")

  # Print the plot for the current gene on a new page
  print(p)
}

# Close the PDF file
dev.off()
```

```{r}
ea_tnf_interaction_genes <- subset(EATNF_list, pvalue <= 0.001)$ensembl_gene_id_version

vsd_counts_melted_selectedgenes <- subset(vsdmelted, ensembl_gene_id_version %in% ea_tnf_interaction_genes)

vsd_counts_melted_selectedgenes_annotated <- merge(vsd_counts_melted_selectedgenes, coldata, by="Sample_ID")

vsd_counts_melted_selectedgenes_annotated$Sample_Group <- factor(vsd_counts_melted_selectedgenes_annotated$Sample_Group, levels =
                                                                   c("WT_TNF_UT", "WT_TNF_1H",
                                                                     "EA_TNF_UT","EA_TNF_1H",
                                                                     "CAEA_TNF_UT", "CAEA_TNF_1H"))

vsd_counts_melted_selectedgenes_annotated <- vsd_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)

#Not actually subsetting here because we are choosing the entire dataset but this is how to subset in case if it is needed.
vsd_counts_melted_selectedgenes_annotated_subset <- subset(vsd_counts_melted_selectedgenes_annotated, Sample_Group %in%
                                                             c("WT_TNF_UT", "WT_TNF_1H",
                                                               "EA_TNF_UT","EA_TNF_1H",
                                                               "CAEA_TNF_UT", "CAEA_TNF_1H"))


# Start a new PDF file with 1 plot per page
pdf("TNFdataset/gene_boxplots_EA_TNF_Interaction.pdf", width = 1.8, height=3)

# Loop over each gene and create a plot on a new page
for (g in unique(vsd_counts_melted_selectedgenes_annotated_subset$ensembl_gene_id_version)) {
  # Subset the data for the current gene
  gene_data <- subset(vsd_counts_melted_selectedgenes_annotated_subset, ensembl_gene_id_version == g)
  # Create the plot for the current gene
  p <- ggplot(gene_data, aes(x = Sample_Group, y = Count, color = Sample_Group)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(width = 0.1) +
    scale_color_manual(values=c("WT_TNF_UT" = "#B0B0B0",
                  "WT_TNF_1H" = "#696969",
                  "EA_TNF_UT" = "#ADD8E6",
                  "EA_TNF_1H" = "#4682B4",
                  "CAEA_TNF_UT" = "#DDA0DD",
                  "CAEA_TNF_1H" = "#800080")) +
    ylab("Normalized Count") +
    ggtitle(paste(gene_data$unique_gene_id)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          plot.background = element_blank(),
          axis.line = element_line(),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
          plot.title = element_text(size=6),
          axis.title.x = element_blank(),
          axis.title.y = element_text(size=10))+
    # theme(axis.title.x = element_blank(),
    #       axis.ticks.x = element_blank(),
    #       strip.background = element_blank(),
    #       strip.placement = "outside",
    #       strip.text = element_text(size = 10, face = "bold"),
    #       axis.text.x = element_text(angle = 90))+
    guides(color="none")

  # Print the plot for the current gene on a new page
  print(p)
}

# Close the PDF file
dev.off()
```

### GSEA ###

```{r}
gsea_df <- subset(gsva_df, gs_name == "TNF_UP_SIGNATURE")
```

```{r}
# WT GSEA
WT_TNFvsUT_ranked<-WT_TNFvsUT_list$rank
names(WT_TNFvsUT_ranked)<-str_remove(WT_TNFvsUT_list$ensembl_gene_id_version, "\\..*$")

enrich_WT_TNFvsUT_GSEAobject <- GSEA(WT_TNFvsUT_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,
                                          nPermSimple = 1e4)
enrich_WT_TNFvsUT_GSEAobject <- setReadable(enrich_WT_TNFvsUT_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_WT_TNFvsUT <- enrich_WT_TNFvsUT_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```
```{r}
# EA GSEA
EA_TNFvsUT_ranked<-EA_TNFvsUT_list$rank
names(EA_TNFvsUT_ranked)<-str_remove(EA_TNFvsUT_list$ensembl_gene_id_version, "\\..*$")

enrich_EA_TNFvsUT_GSEAobject <- GSEA(EA_TNFvsUT_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,
                                          nPermSimple = 1e4)
enrich_EA_TNFvsUT_GSEAobject <- setReadable(enrich_EA_TNFvsUT_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_EA_TNFvsUT <- enrich_EA_TNFvsUT_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```

```{r}
# CAEA GSEA
CAEA_TNFvsUT_ranked<-CAEA_TNFvsUT_list$rank
names(CAEA_TNFvsUT_ranked)<-str_remove(CAEA_TNFvsUT_list$ensembl_gene_id_version, "\\..*$")

enrich_CAEA_TNFvsUT_GSEAobject <- GSEA(CAEA_TNFvsUT_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,
                                          nPermSimple = 1e4)
enrich_CAEA_TNFvsUT_GSEAobject <- setReadable(enrich_CAEA_TNFvsUT_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_CAEA_TNFvsUT <- enrich_CAEA_TNFvsUT_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```
```{r}
# TNF treated CAEA vs EA GSEA
TNF_CAEAvsEA_ranked<-TNF_CAEAvsEA_list$rank
names(TNF_CAEAvsEA_ranked)<-str_remove(TNF_CAEAvsEA_list$ensembl_gene_id_version, "\\..*$")

enrich_TNF_CAEAvsEA_GSEAobject <- GSEA(TNF_CAEAvsEA_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,                                          
                                          pvalueCutoff = 1,
                                          nPermSimple = 1e4)
enrich_TNF_CAEAvsEA_GSEAobject <- setReadable(enrich_TNF_CAEAvsEA_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_TNF_CAEAvsEA <- enrich_TNF_CAEAvsEA_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```

```{r}
# TNF treated CAEA vs WT GSEA
TNF_CAEAvsWT_ranked<-TNF_CAEAvsWT_list$rank
names(TNF_CAEAvsWT_ranked)<-str_remove(TNF_CAEAvsWT_list$ensembl_gene_id_version, "\\..*$")

enrich_TNF_CAEAvsWT_GSEAobject <- GSEA(TNF_CAEAvsWT_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,
                                          pvalueCutoff = 1,
                                          nPermSimple = 1e4)
enrich_TNF_CAEAvsWT_GSEAobject <- setReadable(enrich_TNF_CAEAvsWT_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_TNF_CAEAvsWT <- enrich_TNF_CAEAvsWT_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```

```{r}
# TNF treated EA vs WT GSEA
TNF_EAvsWT_ranked<-TNF_EAvsWT_list$rank
names(TNF_EAvsWT_ranked)<-str_remove(TNF_EAvsWT_list$ensembl_gene_id_version, "\\..*$")

enrich_TNF_EAvsWT_GSEAobject <- GSEA(TNF_EAvsWT_ranked, 
                                          TERM2GENE = gsea_df, 
                                          eps = 0,
                                          pvalueCutoff = 1,
                                          nPermSimple = 1e4)
enrich_TNF_EAvsWT_GSEAobject <- setReadable(enrich_TNF_EAvsWT_GSEAobject, 'org.Hs.eg.db', 'ENSEMBL')
enrich_TNF_EAvsWT <- enrich_TNF_EAvsWT_GSEAobject@result %>%
  dplyr::mutate(signed_padj = sign(NES)*-log10(p.adjust)) %>%
  dplyr::arrange(desc(signed_padj))

#saveRDS(reac_enrich_wt_TNFvsUT_GSEAobject, "TNFdataset/GSEA/reac_enrich_wt_TNFvsUT_GSEAobject.RDS")
```

### GSEA VISUALIZATIONS ###

```{r}
geneSetID_WT_TNFvsUT = match(unique(gsea_df$gs_name), 
                                 enrich_WT_TNFvsUT_GSEAobject@result[["ID"]])
geneSetID_EA_TNFvsUT = match(unique(gsea_df$gs_name), 
                                 enrich_EA_TNFvsUT_GSEAobject@result[["ID"]])
geneSetID_CAEA_TNFvsUT = match(unique(gsea_df$gs_name), 
                                 enrich_CAEA_TNFvsUT_GSEAobject@result[["ID"]])
geneSetID_TNF_EAvsWT = match(unique(gsea_df$gs_name), 
                                 enrich_TNF_EAvsWT_GSEAobject@result[["ID"]])
geneSetID_TNF_CAEAvsEA = match(unique(gsea_df$gs_name), 
                                 enrich_TNF_CAEAvsEA_GSEAobject@result[["ID"]])
geneSetID_TNF_CAEAvsWT = match(unique(gsea_df$gs_name), 
                                 enrich_TNF_CAEAvsWT_GSEAobject@result[["ID"]])

pdf("TNFdataset/GSEA/gseaplot2_WT_TNFvsUT.pdf", width=8, height=4.5)
gseaplot2(enrich_WT_TNFvsUT_GSEAobject, geneSetID = geneSetID_WT_TNFvsUT[!is.na(geneSetID_WT_TNFvsUT)], 
          pvalue_table = TRUE, 
          title="WT TNF vs UT",
          rel_heights = c(2,0.3,0.5))
dev.off()

pdf("TNFdataset/GSEA/gseaplot2_EA_TNFvsUT.pdf", width=8, height=4.5)
gseaplot2(enrich_EA_TNFvsUT_GSEAobject, geneSetID = geneSetID_EA_TNFvsUT[!is.na(geneSetID_EA_TNFvsUT)], 
          pvalue_table = TRUE, 
          title="EA TNF vs UT",
          rel_heights = c(2,0.3,0.5))
dev.off()

pdf("TNFdataset/GSEA/gseaplot2_CAEA_TNFvsUT.pdf", width=8, height=4.5)
gseaplot2(enrich_CAEA_TNFvsUT_GSEAobject, geneSetID = geneSetID_CAEA_TNFvsUT[!is.na(geneSetID_CAEA_TNFvsUT)], 
          pvalue_table = TRUE, 
          title="CAEA TNF vs UT",
          rel_heights = c(2,0.3,0.5))
dev.off()

pdf("TNFdataset/GSEA/gseaplot2_TNF_EAvsWT.pdf", width=8, height=4.5)
gseaplot2(enrich_TNF_EAvsWT_GSEAobject, geneSetID = geneSetID_TNF_EAvsWT[!is.na(geneSetID_TNF_EAvsWT)], 
          pvalue_table = TRUE, 
          title="TNF treated EA vs WT",
          rel_heights = c(2,0.3,0.5))
dev.off()

pdf("TNFdataset/GSEA/gseaplot2_TNF_CAEAvsWT.pdf", width=8, height=4.5)
gseaplot2(enrich_TNF_CAEAvsWT_GSEAobject, geneSetID = geneSetID_TNF_CAEAvsWT[!is.na(geneSetID_TNF_CAEAvsWT)], 
          pvalue_table = TRUE, 
          title="TNF treated CAEA vs WT",
          rel_heights = c(2,0.3,0.5))
dev.off()

pdf("TNFdataset/GSEA/gseaplot2_TNF_CAEAvsEA.pdf", width=8, height=4.5)
gseaplot2(enrich_TNF_CAEAvsEA_GSEAobject, geneSetID = geneSetID_TNF_CAEAvsEA[!is.na(geneSetID_TNF_CAEAvsEA)], 
          pvalue_table = TRUE, 
          title="TNF treated CAEA vs EA",
          base_size = 16,
          rel_heights = c(2,0.3,0.5))
dev.off()
```



